pkg/service/http_service.go

```diff
@@ -1,60 +1,63 @@
 package service
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
 	"net/http"
+
+	gen "github.com/open-feature/flagd/pkg/generated"
 )
 
 type HttpServiceConfiguration struct {
 	Port int32
 }
 
-type HttpServiceRequest struct {
-	Payload string
+type HttpService struct {
+	HttpServiceConfiguration *HttpServiceConfiguration
 }
 
-type HttpServiceResponse struct {
-	rawPayload string
-}
+var defaultReason = "DEFAULT"
 
-func (h *HttpServiceResponse) GetPayload() string {
-	return h.rawPayload
+// implement the generated ServerInterface.
+type Server struct {}
```

Comment:
The code generator defines an interface that we must implement to handle our request. It also generates middlewares that do the parsing and validation, which is in the generated code that should not be manually modified (`pk/generated/server.gen.go`